src/table/use-column-widths.tsx

```diff
@@ -80,6 +82,36 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, childre
     }
   };
 
+  // Update getContainer when deps change to trigger new observation.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getContainer = useCallback(() => containerRef.current, [resizableColumns, visibleColumns, columnWidths]);
+
+  // Measure wrapper width to determine if the last column can stretch.
+  // When condition changes, updates the last column width style imperatively to avoid having a frame with improper style.
+  const [containerWidth, setContainerWidth] = useState(0);
+  useResizeObserver(getContainer, entry => {
+    if (!resizableColumns) {
+      return;
+    }
+
+    setContainerWidth(entry.contentBoxWidth);
+
+    const lastColumnId = visibleColumns[visibleColumns.length - 1]?.id;
+    const lastCellElement = lastColumnId ? cellsRef.current[lastColumnId] : null;
+    if (!lastCellElement) {
```

Comment:
How is this possible? How can you have a table without the last column?

Reply:
It is technically possible to have a table with no columns. This can be the case for example when the columns configuration is resolved asynchronously.

Reply:
👀 it is first time I hear about this use-case. Do we have any unit tests confirming that our table works in this setup?

Reply:
Surprisingly, I haven't found tests or permutations for that case, but the component does work when passing an empty column definitions array.

Reply:
All right, something to think about later. For this story I am wondering if we can make even earlier return, together with `!resizableColumns`, so even the container width is not tracked

Technically, if there are no columns, there are no resizable columns too 🤷

Reply:
Makes sense!